.github/workflows/ci.yml

```diff
@@ -0,0 +1,47 @@
+name: CI
+on: [push, pull_request]
+env:
+  CI: true
+
+jobs:
+  test:
+    runs-on: ${{ matrix.os }}
+
+    strategy:
+      fail-fast: false
+      matrix:
+        python: [3.5, 3.6, 3.7]
+        os: [ubuntu-latest, macos-latest]
+
+    steps:
+      - name: Clone repository
+        uses: actions/checkout@v2
+
+      - name: Set up Python
+        uses: actions/setup-python@v1
+        with:
+          python-version: ${{ matrix.python }}
+          architecture: 'x64'
+
+      - run: python --version
+
+      - name: Cache dependencies
+        uses: actions/cache@v1
+        with:
+          path: ~/miniconda3
+          key: ${{ runner.os }}-python-v${{ matrix.python }}-${{ hashFiles('ci/*.sh') }}-${{ hashFiles('.github/workflows/ci.yml') }}
+          restore-keys: ${{ runner.os }}-python-v${{ matrix.python }}-${{ hashFiles('ci/*.sh') }}-${{ hashFiles('.github/workflows/ci.yml') }}
+
+      - name: Install conda
+        env:
+          PYTHON_VERSION: ${{ matrix.python }}
+        run: |
+          export PATH="$HOME/miniconda3/bin:$PATH"
+          ci/install_conda.sh
+          ci/setup_conda_env.sh
+
+      - name: Run tests
+        run: |
+          export PATH="$HOME/miniconda3/bin:$PATH"
+          ci/test.sh
+          ci/lint.sh
```

Comment:
Linting (`ci/lint.sh`) should have it's own "jobs"/"stage" (don't know the word yet 😥).

And it should be done before the test run and not after. :smile_cat: I know it was done like that before, but it does not make sense to lint after the tests ...

Reply:
If you check the current CI Travis workflow, you'll see it's like that. This PR just tries to stay as close as the current Travis CI as possible.

So, I won't make this change nor the Python 3.8 addition here. Those should be made separately.